File: Lecture022 Strings/reverseString.cpp
#include<string>
#include<iostream>
#include<vector>
using namespace std;
class Solution {
public:

    void reverseString(vector<char>& s) {
        int st=0;
        int e = s.size()-1;

        while(st<e) {
            swap(s[st++], s[e--]);
        }
    }
};

File: Lecture022 Strings/reverseString.java
class Solution {
    public void reverseString(char[] s) {
        //using two pointers method
        int leftPointer =0;
        int rightPointer=s.length-1;
        
        while(leftPointer<rightPointer){
            char temp = s[leftPointer];
            s[leftPointer++]=s[rightPointer];
            s[rightPointer--]=temp;
        }
    }
}
       
